micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -107,11 +104,16 @@ public <T> T timeControllerExecution(ControllerExecution<T> execution) {
     final var resourceID = execution.resourceID();
     final var metadata = execution.metadata();
     final var tags = new ArrayList<String>(metadata.size() + 4);
-    tags.addAll(List.of(
-        "controller", name,
-        "resource.name", resourceID.getName(),
-        "resource.namespace", resourceID.getNamespace().orElse(""),
-        "resource.scope", getScope(resourceID)));
+    if (collectPerResourceMetrics) {
+      tags.addAll(List.of(
+          "controller", name,
+          "resource.name", resourceID.getName(),
+          "resource.namespace", resourceID.getNamespace().orElse(""),
+          "resource.scope", getScope(resourceID)));
+    } else {
+      tags.add("controller");
+      tags.add(name);
+    }
```

Comment:
```suggestion
    tags.add("controller");
    tags.add(name);
    if (collectPerResourceMetrics) {
      tags.addAll(List.of(
          "resource.name", resourceID.getName(),
          "resource.namespace", resourceID.getNamespace().orElse(""),
          "resource.scope", getScope(resourceID)));
    }
```
I would move the per resource block after metadata.